pkg/instrumentation/sdk.go

```diff
@@ -424,7 +425,7 @@ func (i *sdkInjector) createResourceMap(ctx context.Context, otelinst v1alpha1.I
 			res[k] = v
 		}
 	}
-
+	serviceInstanceId, _ := os.Hostname()
```

Comment:
If an error is returned we can ignore it, but I don't think we should set the serviceInstanceId anymore.

Reply:
Do you have any suggestion for values to be set as service.instance.id?

Reply:
I propose we only set if it is retrievable.  If `os.Hostname()` returns an error we should not set it.

Reply:
Does `os.Hostname()` guarantee the requirement?

>_service.namespace,service.name,service.instance.id triplet MUST be globally unique_

Collector uses uuid following the semconv guideline 

>_If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations)_

Reply:
ok, so I basically created a new UUID and implemented it as a service.instance.id.

Reply:
I think we can do better than generating a new UUID. We should consider using the pod UID which should be unique.

Reply:
Replaced to the pod.UID.